### PR TITLE
Issue #9749: update MethodLengthCheckTest.java to use unique Input file in each test method

### LINKED ISSUE #9749: update MethodLengthCheckTest.java to use unique Input file in each test method
MethodLengthCheckTest.java file reuse some Input files few time in few different test method.
Expected: each test method in Test class use its own Input file and do not share it with any other test method.

more details - https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-create-Input-files-for-tests

### PR DESCRIPTION
**Pull request by MahmoudAbdelazim**
Closes #9749 

I separated the input file that was shared between two test methods inside MethodLengthCheckTest.java file, and I added a // violation comment next to the violation test method.

But the file (InputMethodLengthSimple) contains many things that I don't think is related to the Method Length functionality, it contains things like name format tests, accessor tests, local variables tests, and many others.

So if that should be fixed, should I create an issue with this problem and try to fix, or it's intended to be that way?
thank you in advance.

### CODE REVIEW

**Comment by MahmoudAbdelazim** [comment]
@romani Can you please review the changes?

File: config/checkstyle_input_suppressions.xml [hunk 1]
```diff
@@ -1582,8 +1582,6 @@
             files="sizes[\\/]methodlength[\\/]InputMethodLengthComments.java"/>
   <suppress id="ConfigCommentOnTopInputs"
             files="sizes[\\/]methodlength[\\/]InputMethodLengthModifier.java"/>
```
Comment (timurt) [inline config/checkstyle_input_suppressions.xml:1584]: please remove all suppressions for `methodlength` inputs

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/sizes/MethodLengthCheckTest.java [hunk 2]
```diff
@@ -75,7 +75,7 @@ public void testCountEmpty() throws Exception {
         checkConfig.addAttribute("max", "19");
         checkConfig.addAttribute("countEmpty", "false");
         final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
-        verify(checkConfig, getPath("InputMethodLengthSimple.java"), expected);
+        verify(checkConfig, getPath("InputMethodLengthCountEmpty.java"), expected);
```
Comment (timurt) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/sizes/MethodLengthCheckTest.java:78]: it seems like name of the test contradicts to what it does: `testCountEmpty` and `countEmpty = false`

lets change test name and input name to
`testCountEmptyIsFalse`  `InputMethodLengthCountEmptyIsFalse`

what do you think?

**Review by timurt — CHANGES_REQUESTED** [review]

File: config/checkstyle_input_suppressions.xml (hunk 1 above)
Comment (MahmoudAbdelazim) [reply config/checkstyle_input_suppressions.xml:1584]: On it.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/sizes/MethodLengthCheckTest.java (hunk 2 above)
Comment (MahmoudAbdelazim) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/sizes/MethodLengthCheckTest.java:78]: On it.

File: config/checkstyle_input_suppressions.xml (hunk 1 above)
Comment (MahmoudAbdelazim) [reply config/checkstyle_input_suppressions.xml]: Done.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/sizes/MethodLengthCheckTest.java (hunk 2 above)
Comment (MahmoudAbdelazim) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/sizes/MethodLengthCheckTest.java]: Done.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodlength/InputMethodLengthModifier.java [hunk 3]
```diff
@@ -5,6 +5,9 @@
 
 package com.puppycrawl.tools.checkstyle.checks.sizes.methodlength;
 
+/*
+ * Config:
```
Comment (timurt) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodlength/InputMethodLengthModifier.java:9]:
```
/*
 * Config: default
*/
```

**Review by timurt — CHANGES_REQUESTED** [review]

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodlength/InputMethodLengthModifier.java (hunk 3 above)
Comment (MahmoudAbdelazim) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodlength/InputMethodLengthModifier.java]: Done.

**Comment by MahmoudAbdelazim** [comment]
@timurt please review the changes.

**Comment by MahmoudAbdelazim** [comment]
@romani

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodlength/InputMethodLengthCountEmptyIsFalse.java [hunk 4]
```diff
@@ -0,0 +1,230 @@
+////////////////////////////////////////////////////////////////////////////////
```
Comment (timurt) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodlength/InputMethodLengthCountEmptyIsFalse.java:1]: @romani @pbludov I am not sure, should we include such headers in newly created input files?

Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodlength/InputMethodLengthCountEmptyIsFalse.java:1]: @MahmoudAbdelazim , please remove such header
and remove whole comment "Contains simple mistakes: ......"

Comment (MahmoudAbdelazim) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodlength/InputMethodLengthCountEmptyIsFalse.java]: @romani @timurt Done.

**Comment by MahmoudAbdelazim** [comment]
@romani @timurt Are there any other changes requested?

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodlength/InputMethodLengthSimple.java [hunk 5]
```diff
@@ -1,18 +1,10 @@
...
- * - Long lines
- * - Tabs
- * - Format of variables and parameters
- * - Order of modifiers
- * @author Oliver Burn
- **/
+/*
+ * Config:
+ * max = 19
+ * countEmpty = true
```
Comment (timurt) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodlength/InputMethodLengthSimple.java:6]: last comment: you can remove this line as `countEmpty  = true` by default and wasn't set explicitely inside test configuration

**Review by timurt — CHANGES_REQUESTED** [review]

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodlength/InputMethodLengthSimple.java (hunk 5 above)
Comment (MahmoudAbdelazim) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodlength/InputMethodLengthSimple.java:6]: That's right, I'll do it.

Comment (MahmoudAbdelazim) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodlength/InputMethodLengthSimple.java]: Done.

**Comment by MahmoudAbdelazim** [comment]
@timurt All changes done.

**Pull request merged**